src/Gren/Kernel/Crypto.js

```diff
@@ -0,0 +1,660 @@
+        case 192:
+            aesKeyData.__$length = __Crypto_AesLength192
+        case 256:
+            aesKeyData.__$length = __Crypto_AesLength256
+    }
+    return A2(__Crypto_Key, key, aesKeyData);
+};
+
+var _Crypto_constructEcKey = function (key) {
+    var ecKeyData = {
+        __$extractable: _Crypto_extractableFromBool(key.extractable)
+    };
+    switch (key.algorithm.namedCurve) {
+        case "P-256":
+            ecKeyData.__$namedCurve = __Crypto_P256;
+        case "P-384":
+            ecKeyData.__$namedCurve = __Crypto_P384;
+        case "P-521":
+            ecKeyData.__$namedCurve = __Crypto_P521;
+    }
+    return A2(__Crypto_Key, key, ecKeyData);
+};
+
+// Random
+
+var _Crypto_randomUUID = __Scheduler_binding(function (callback) {
+    var randomUUID = crypto.randomUUID();
+    return callback(__Scheduler_succeed(randomUUID));
+});
+
+var _Crypto_getRandomValues = F2(function (arrayLength, valueType) {
+    var array;
+    switch (valueType) {
+        case "int8":
+            array = new Int8Array(arrayLength);
+            break;
+        case "uint8":
+            array = new Uint8Array(arrayLength);
+            break;
+        case "int16":
+            array = new Int16Array(arrayLength);
+            break;
+        case "uint16":
+            array = new Uint16Array(arrayLength);
+            break;
+        case "int32":
+            array = new Int32Array(arrayLength);
+            break;
+        case "uint32":
+            array = new Uint32Array(arrayLength);
+            break;
+        default:
+            array = new Int8Array(0);
+            break;
+    }
+    var randomValues = crypto.getRandomValues(array);
+    return __Scheduler_binding(function (callback) {
+        return callback(__Scheduler_succeed(new DataView(randomValues.buffer)));
+    })
+});
+
+// Context
+
+var _Crypto_getContext = __Scheduler_binding(function (callback) {
+    if (crypto.subtle) {
+        return callback(__Scheduler_succeed(__Crypto_SecureContext));
+    }
+    return callback(__Scheduler_fail());
+});
+
+// Generate keys
+
+var _Crypto_generateRsaKey = F6(function (name, modulusLength, publicExponent, hash, extractable, permissions) {
+    var algorithm = {
+        name: name,
+        modulusLength: modulusLength,
+        publicExponent: new Uint8Array(publicExponent),
+        hash: hash
+    }
+    return __Scheduler_binding(function (callback) {
+        crypto.subtle
+            .generateKey(algorithm, extractable, permissions)
+            .then(function (key) {
+                return callback(__Scheduler_succeed(
+                    {
+                        __$publicKey: __Crypto_PublicKey(_Crypto_constructRsaKey(key.publicKey)),
+                        __$privateKey: __Crypto_PrivateKey(_Crypto_constructRsaKey(key.privateKey))
+                    }
+                ))
+            }).catch(function (err) {
+                return callback(__Scheduler_fail());
```

Comment:
You're calling `__Scheduler_fail()` without an argument in several places. I assume this is because you're certain that the function will never fail in practice.

I wonder if it would be better to throw an actual exception instead. If it never happens, then it doesn't matter, but if it does happen we at least will discover the bug quickly. The exception should include a message urging the user to submit a ticket.